dial.go

```diff
@@ -27,34 +39,26 @@ func (d contextDialer) DialContext(ctx context.Context, network, addr string, co
 		return nil, err
 	}
 
-	type dialRes struct {
-		client *ssh.Client
-		err    error
-	}
-	dialc := make(chan dialRes, 1)
-
+	var client *ssh.Client
```

Comment:
I fail to understand purpose of this change. This function in design and naming follows http package no need to butcher it.